core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,444 @@
+          stat.language_code);
+        // This throws "Object is possibly 'undefined'.".
+        // We are checking whether there exists translation review stats
+        // data for a given language. If there are no stats data or it is
+        // undefined, we are setting it here. Hence we can ignore the ts check
+        // here. We need to suppress this error because of strict type checking.
+        // @ts-ignore
+        if (!this.statsData.translationReview[language]) {
+          // This throws "Object is possibly 'undefined'.".
+          // We are checking whether there exists translation review stats
+          // data for a given language. If there are no stats data or it is
+          // undefined, we are setting it here. Hence we can ignore the ts check
+          // here. We need to suppress this error because of strict type
+          // checking.
+          // @ts-ignore
+          this.statsData.translationReview[language] = this
+            .createTranslationReviewPageableStats(stat);
+        } else {
+          // This throws "Object is possibly 'undefined'.".
+          // In the previous if block, we are checking whether
+          // statsData.translationReview[language] is undefined. Hence we can
+          // confirm that statsData.translationReview[language] is not undefined
+          // in this block. Hence we can ignore the ts check
+          // here. We need to suppress this error because of strict type
+          // checking.
+          // @ts-ignore
+          this.statsData?.translationReview[language].data.push(
+            this.createTranslationReviewStat(stat));
+        }
+      });
+    }
+
+    if (response.question_contribution_stats.length > 0) {
+      this.statsData.questionContribution = this
+        .createQuestionContributionPageableStats(
+          response.question_contribution_stats);
+    }
+
+    if (response.question_review_stats.length > 0) {
+      this.statsData.questionReview = this.createQuestionReviewPageableStats(
+        response.question_review_stats);
+    }
+  }
+
+  createTranslationContributionPageableStats(
+      stat: TranslationContributionBackendDict): PageableStats {
+    const data = [this.createTranslationContributionStat(stat)];
+    return new PageableStats(data);
+  }
+
+  createTranslationReviewPageableStats(
+      stat: TranslationReviewBackendDict): PageableStats {
+    const data = [this.createTranslationReviewStat(stat)];
+    return new PageableStats(data);
+  }
+
+  createQuestionContributionPageableStats(
+      stat: QuestionContributionBackendDict[]): PageableStats {
+    const data = stat.map((stat) => {
+      return this.createQuestionContributionStat(stat);
+    });
+    return new PageableStats(data);
+  }
+
+  createQuestionReviewPageableStats(
+      stat: QuestionReviewBackendDict[]): PageableStats {
+    const data = stat.map((stat) => {
+      return this.createQuestionReviewStat(stat);
+    });
+    return new PageableStats(data);
+  }
```

Comment:
Ah, sorry in my previous comment I meant we can delete these functions and just create the Pageable stats inline above.

Reply:
Done